iox_catalog/src/interface.rs

```
@@ -369,7 +359,7 @@ pub trait ColumnRepo: Send + Sync {
     async fn create_or_get_many_unchecked(
         &mut self,
         table_id: TableId,
-        columns: &[ColumnUpsertRequest<'_>],
+        columns: HashMap<&str, ColumnType>,
```

what was the rationale for using a HashMap here? Is it so that the validation after catalog insert could be made faster?

```rust
            let want = columns.get(existing.name.as_str()).unwrap();
```

I was just wondering if we had somehow masked the error by simply changing the order of columns passed to the SQL statement (as now it is the hash of the column name rather than the order in the write) 🤔

---

I tried to document my rationale in the 2nd commit's message, does this explain it enough?

> The checks for whether a column already exists with a different type
were relying on ordering of the input matching the ordering of the
columns returned from inserting the columns in Postgres.
>
>Rather than trying to match the new ordering that is required to avoid
Postgres deadlocks, switch from a Vec to a HashMap and look up the
column type from the name.
>
> This also reduces some allocations that weren't really needed.

---

yes, thank you, sorry I do not typically look at messages on individual commits during code review -- I look at the description on the PR and comments in the code